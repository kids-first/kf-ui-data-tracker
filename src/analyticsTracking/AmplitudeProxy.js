import {Amplitude} from '@amplitude/react-amplitude';
import {memoize} from '@amplitude/react-amplitude/src/lib/memoize';
import debounce from 'lodash.debounce';
import {buttonTracking} from './eventUtils';
import {EVENT_CONSTANTS} from '../analyticsTracking';

/**
 *
 */
class AmplitudeProxy extends Amplitude {
  logToConsole = false;

  constructor(props) {
    super(props);
    if (typeof props.debounceInterval === 'number') {
      this.logEvent = debounce(this.dispatch, props.debounceInterval);
    } else {
      this.logEvent = this.dispatch;
    }

    this._renderPropParams = {
      getInstance: this.getAmplitudeInstance,
      logEvent: this.logEvent,
      instrument: this.instrument,
      EVENT_CONSTANTS,
      buttonTracking: buttonTracking(this.logEvent),
    };
  }

  // proxy our logging calls so we can hook
  // other analytics services into it
  dispatch = (eventType, eventProps, cb) => {
    if (this.logToConsole || this.props.logToConsole) {
      console.log(
        `AmplitudeProxy::dispatch eventType:${eventType}`,
        eventProps,
      );
    }

    const combinedEventProps = {
      ...this.getAmplitudeEventProperties(),
      ...(eventProps || {}),
    };

    if (
      process.env.NODE_ENV === 'test' ||
      process.env.NODE_ENV === 'CI' ||
      this.props.storeSessionEvents
    ) {
      /** log all events to sessionStorage */
      sessionStorage.setItem(
        'session_events',
        JSON.stringify([
          ...(JSON.parse(sessionStorage.getItem('session_events')) || []),
          {
            eventType,

            eventProps: combinedEventProps,
            utc_time: Date.now(),
          },
        ]),
      );
    }

    return this._makeLogEvent()(eventType, combinedEventProps, cb);
  };

  instrument = memoize((eventType, func, props = {}) => {
    return (...params) => {
      const retVal = typeof func === 'function' ? func(...params) : undefined;
      this.dispatch(eventType, props);

      return retVal;
    };
  });
}

export default AmplitudeProxy;
