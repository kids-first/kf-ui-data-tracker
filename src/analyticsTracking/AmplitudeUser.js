import amplitude from 'amplitude-js';
import jwtDecode from 'jwt-decode';

/**
 * convenience class to log user properties on Auth0 login success
 * @see https://help.amplitude.com/hc/en-us/articles/115002380567-User-Properties-Event-Properties#applying-user-properties-to-events
 */
class AmplitudeUser {
  instance;
  userId;
  auth_sub_arr;

  // decoded jwt values
  auth_user;

  // values to pluck from jwt in the format of [[jwt_prop_name, renmae_to], ... ]
  user_props = [
    ['https://kidsfirstdrc.org/roles', 'roles'],
    ['https://kidsfirstdrc.org/groups', 'studies'],
    ['https://kidsfirstdrc.org/permissions', 'permissions'],
  ];

  constructor(idToken, api_key) {
    amplitude.getInstance().init(api_key);
    this.instance = amplitude.getInstance();
    this.auth_user = jwtDecode(idToken);

    this.auth_sub_arr = this.auth_user.sub.split('|');

    this.setId();
    this.setUserProperies();
    return this;
  }

  /**
   * UUID for amplitude using the parsed value of the
   * auth provider sub field
   * @param {string} id
   */
  setId(id) {
    this.userId = this.auth_sub_arr[1];
    this.instance.setUserId(this.userId);
  }

  setUserProperies() {
    var identify = new this.instance.Identify();

    // pluck properties from dceoded jwt and rename
    this.user_props.forEach(prop => {
      identify.set(prop[1] || prop[0], this.auth_user[prop[0]]);
    });
    // send to amplitude
    this.instance.identify(identify);
  }
}

export default AmplitudeUser;
