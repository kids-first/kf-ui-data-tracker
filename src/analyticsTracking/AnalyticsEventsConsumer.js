import React from 'react';
import {Amplitude} from '@amplitude/react-amplitude';
import useAnalyticsTracking from './useAnalyticsTracking';

const AnalyticsEventsConsumer = ({children, eventProps}) => {
  const analyticsConstants = useAnalyticsTracking();

  return (
    <Amplitude
      eventProperties={inheritedProps => ({
        ...inheritedProps,
        ...eventProps,
      })}
    >
      {props => {
        const buttonTrackingProps = buttonName => ({
          onClick: e =>
            props.logEvent(analyticsConstants.MOUSE.CLICK, {
              button_name: buttonName,
            }),
          onMouseEnter: e =>
            props.logEvent(analyticsConstants.MOUSE.HOVER, {
              button_name: buttonName,
            }),
        });
        if (typeof children === 'function') {
          return children({...props, buttonTrackingProps, analyticsConstants});
        } else {
          return children || null;
        }
      }}
    </Amplitude>
  );
};

export default AnalyticsEventsConsumer;
