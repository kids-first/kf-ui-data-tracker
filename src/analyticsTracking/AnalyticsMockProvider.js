import React from 'react';
import AnalyticsTrackingProvider from './AnalyticsTrackingProvider';

import {Amplitude, LogOnMount} from '@amplitude/react-amplitude';

const AnaltyticsMockProvider = ({children}) => (
  <AnalyticsTrackingProvider>
    <Amplitude eventProperties={{scope: ['TEST']}}>
      <LogOnMount eventType={'TEST_MOUNT'} />
      {children}
    </Amplitude>
  </AnalyticsTrackingProvider>
);

export default AnaltyticsMockProvider;
