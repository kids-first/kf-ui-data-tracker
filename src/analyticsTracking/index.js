export {default as useAnalyticsTracking} from './useAnalyticsTracking';
export {default as withAnalyticsTracking} from './withAnalyticsTracking';

export {
  default as AnalyticsTrackingProvider,
} from './AnalyticsTrackingProvider';

export {default as AmplitudeProxy} from './AmplitudeProxy';
export {default as EVENT_CONSTANTS} from '../common/analyticsTrackingConstants';
