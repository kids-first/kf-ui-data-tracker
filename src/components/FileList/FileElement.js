import React, {useState, useEffect} from 'react';
import PropTypes from 'prop-types';
import {withRouter, Link} from 'react-router-dom';
import TimeAgo from 'react-timeago';
import Badge from '../Badge/Badge';
import {Header, Table, Icon, List, Responsive} from 'semantic-ui-react';

import FileActionsContainer from '../../containers/FileActionsContainer';
import {
  fileSortedVersions,
  fileLatestDate,
  fileLatestSize,
  lengthLimit,
} from '../../common/fileUtils';
import {fileTypeDetail} from '../../common/enums';
import {longDate} from '../../common/dateUtils';

const useRecentlyUpdated = (latestDate, fileId) => {
  const [justUpdated, setJustUpdated] = useState(false);

  useEffect(() => {
    let FIVE_MIN = 300000;
    let docs = sessionStorage.getItem('kf_updated_docs');
    // make sure we only select one row
    if (fileId === docs) {
      // make sure that row was updated less than 5 minutes ago
      if (new Date().getTime() - Date.parse(latestDate) < FIVE_MIN) {
        setJustUpdated(true);
        setTimeout(() => {
          // reset after 7seconds
          setJustUpdated(false);
          sessionStorage.removeItem('kf_updated_docs');
        }, 7000);
      }
    }
    return () => {
      setJustUpdated(false);
    };
  }, [latestDate, fileId]);

  return justUpdated;
};

/**
 * Displays a list of file attributes
 */
const FileAttributes = ({
  latestDate,
  fileSize,
  fileType,
  horizontal,
  fileVersions,
  fileKfID,
}) => {
  const justUpdated = useRecentlyUpdated(latestDate, fileKfID);

  return (
    <List horizontal={horizontal} link={!justUpdated}>
      <List.Item>
        <List.Content verticalAlign="middle">
          <Icon
            name={`${fileType.icon || 'question'}`}
            size="small"
            color="grey"
          />
          {fileType.title}
        </List.Content>
      </List.Item>

      <List.Item>
        <List.Content>
          {justUpdated ? <Icon name="refresh" /> : null}
          {latestDate ? (
            <>
              Modified{' '}
              <TimeAgo
                date={latestDate}
                live={false}
                title={longDate(latestDate)}
              />
            </>
          ) : (
            'Unknown time'
          )}
        </List.Content>
      </List.Item>
      <List.Item>
        <List.Content>{fileVersions} versions </List.Content>
      </List.Item>
      <List.Item>
        <List.Content>{fileSize}</List.Content>
      </List.Item>
    </List>
  );
};

FileAttributes.propTypes = {
  /** The kf id of the file */
  fileKfID: PropTypes.string.isRequired,
  /** The date of last modification */
  latestDate: PropTypes.string.isRequired,
  /** The size of the file */
  fileSize: PropTypes.string.isRequired,
  /** The type of file */
  fileType: PropTypes.object.isRequired,
  /** Whether the list should be displayed horizontally */
  horizontal: PropTypes.bool,
};

FileAttributes.defaultProps = {
  horizontal: false,
};

/**
 * Displays unordered study files in list view
 */
const FileElement = ({fileNode, loading, history, match, fileListId}) => {
  const fileKfID = fileNode.kfId || 'unknown ID';
  const fileName = fileNode.name || 'unknown file name';
  const fileDescription = fileNode.description || null;
  const sortedVersions = fileSortedVersions(fileNode);
  const latestDate = fileLatestDate(sortedVersions);
  const fileSize = fileLatestSize(sortedVersions);
  const fileType =
    fileNode && fileTypeDetail[fileNode.fileType]
      ? fileTypeDetail[fileNode.fileType]
      : {title: 'unknown', icon: 'question'};
  const versionState =
    sortedVersions.length > 0 ? sortedVersions[0].node.state : null;

  const justUpdated = useRecentlyUpdated(latestDate, fileKfID);
  return (
    <Table.Row
      style={{backgroundColor: justUpdated ? '#f8ffff' : 'inherit'}}
      data-testid="file-item"
      className="cursor-pointer"
      onClick={() =>
        history.push(`/study/${match.params.kfId}/documents/${fileKfID}`)
      }
    >
      <Table.Cell textAlign="center">
        <Header>
          {justUpdated && (
            <>
              <Badge state="UPD" icon="refresh" className="mb-5" />
              <br />
            </>
          )}

          <Badge
            state={versionState}
            loading={loading}
            filled={versionState === 'CHN'}
          />
        </Header>
      </Table.Cell>
      <Table.Cell>
        <Header size="medium" as="span">
          <Link to={`/study/${match.params.kfId}/documents/${fileKfID}`}>
            {fileName}
          </Link>
        </Header>
        <p className="noMargin">
          {fileDescription ? <>{lengthLimit(fileDescription, 100)}</> : null}
        </p>
        <Responsive
          as={FileAttributes}
          minWidth={Responsive.onlyTablet.minWidth}
          fileKfID={fileKfID}
          latestDate={latestDate}
          fileSize={fileSize}
          fileType={fileType}
          fileVersions={fileNode.versions.edges.length}
          horizontal={true}
        />
        <Responsive
          as={FileAttributes}
          maxWidth={Responsive.onlyTablet.minWidth}
          fileKfID={fileKfID}
          latestDate={latestDate}
          fileSize={fileSize}
          fileType={fileType}
          fileVersions={fileNode.versions.edges.length}
          horizontal={false}
        />
      </Table.Cell>
      <Table.Cell textAlign="center">
        <Responsive
          as={FileActionsContainer}
          minWidth={Responsive.onlyTablet.minWidth}
          node={fileNode}
          studyId={fileListId}
          vertical={true}
          fluid={false}
        />
        <Responsive
          as={FileActionsContainer}
          maxWidth={Responsive.onlyTablet.minWidth}
          node={fileNode}
          studyId={fileListId}
          vertical={false}
          fluid={true}
        />
      </Table.Cell>
    </Table.Row>
  );
};

FileList.propTypes = {
  /** Array of study object*/
  fileNode: PropTypes.object,
};

FileList.defaultProps = {
  fileNode: null,
};

export default withRouter(FileElement);
