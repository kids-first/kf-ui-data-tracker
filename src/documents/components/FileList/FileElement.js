import React, {useState, useEffect} from 'react';
import PropTypes from 'prop-types';
import {withRouter, Link} from 'react-router-dom';
import TimeAgo from 'react-timeago';
import {Table, Icon, Checkbox, Popup} from 'semantic-ui-react';
import Markdown from 'react-markdown';
import FileActionButtons from '../FileActionButtons/FileActionButtons';
import {fileSortedVersions, fileLatestDate} from '../../utilities';
import {fileTypeDetail} from '../../../common/enums';
import {longDate} from '../../../common/dateUtils';
import FileTags from '../FileDetail/FileTags';

const useRecentlyUpdated = (latestDate, fileId) => {
  const [justUpdated, setJustUpdated] = useState(false);

  useEffect(() => {
    let FIVE_MIN = 300000;
    let docs = sessionStorage.getItem('kf_updated_docs');
    // make sure we only select one row
    if (fileId === docs) {
      // make sure that row was updated less than 5 minutes ago
      if (new Date().getTime() - Date.parse(latestDate) < FIVE_MIN) {
        setJustUpdated(true);
        setTimeout(() => {
          // reset after 7seconds
          setJustUpdated(false);
          sessionStorage.removeItem('kf_updated_docs');
        }, 7000);
      }
    }
    return () => {
      setJustUpdated(false);
    };
  }, [latestDate, fileId]);

  return justUpdated;
};

/**
 * Displays unordered study files in list view
 */
const FileElement = ({
  fileNode,
  loading,
  history,
  match,
  fileListId,
  updateFile,
  deleteFile,
  downloadFileMutation,
  selected,
  onSelectOne,
}) => {
  const fileKfID = fileNode.kfId || 'unknown ID';
  const fileName = fileNode.name || 'unknown file name';
  // Show at most 5 lines of the description
  const fileDescription =
    fileNode.description
      .split('\n')
      .splice(0, 5)
      .join('\n') || '';
  const sortedVersions = fileSortedVersions(fileNode);
  const latestDate = fileLatestDate(sortedVersions);
  const fileType =
    fileNode && fileTypeDetail[fileNode.fileType]
      ? fileTypeDetail[fileNode.fileType]
      : {title: 'unknown', icon: 'question'};
  const justUpdated = useRecentlyUpdated(latestDate, fileKfID);
  return (
    <Table.Row
      style={{backgroundColor: justUpdated ? '#f8ffff' : 'inherit'}}
      data-testid="file-item"
      className="cursor-pointer"
      onClick={() =>
        history.push(`/study/${match.params.kfId}/documents/${fileKfID}`)
      }
    >
      <Table.Cell
        textAlign="center"
        onClick={e => {
          e.stopPropagation();
          onSelectOne(fileKfID);
        }}
      >
        <Checkbox data-testid="file-select" checked={selected} />
      </Table.Cell>
      <Popup
        wide="very"
        mouseEnterDelay={500}
        header={fileType.title || 'Unknown Type'}
        content={fileType.description || 'Unknown Type'}
        trigger={
          <Table.Cell textAlign="center">
            <Icon name={fileType.icon || 'question'} />
          </Table.Cell>
        }
      />
      <Table.Cell>
        {fileDescription.length > 3 && (
          <Popup
            offset="-75%"
            wide="very"
            position="top left"
            trigger={<Icon name="info circle" />}
            content={
              <Markdown
                source={fileDescription}
                renderers={{
                  image: Image,
                  table: props => <Table>{props.children}</Table>,
                }}
              />
            }
          />
        )}
        <Link to={`/study/${match.params.kfId}/documents/${fileKfID}`}>
          {fileName}
        </Link>
      </Table.Cell>
      <Table.Cell textAlign="center" width="4">
        <FileTags fileNode={fileNode} updateFile={updateFile} />
      </Table.Cell>
      <Table.Cell textAlign="center" width="1">
        <TimeAgo date={latestDate} live={false} title={longDate(latestDate)} />
      </Table.Cell>
      <Table.Cell textAlign="center">
        <FileActionButtons
          fluid
          node={fileNode}
          studyId={fileListId}
          deleteFile={deleteFile}
          downloadFileMutation={downloadFileMutation}
        />
      </Table.Cell>
    </Table.Row>
  );
};

FileList.propTypes = {
  /** Array of study object*/
  fileNode: PropTypes.object,
};

FileList.defaultProps = {
  fileNode: null,
};

export default withRouter(FileElement);
