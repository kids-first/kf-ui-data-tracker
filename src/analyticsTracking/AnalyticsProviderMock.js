import React from 'react';
import AnalyticsTrackingProvider from './AnalyticsTrackingProvider';
import AmplitudeProxy from './services/Amplitude/AmplitudeProxy';

const AnaltyticsMockProvider = ({children}) => (
  <AnalyticsTrackingProvider>
    <AmplitudeProxy eventProperties={{scope: ['TEST']}}>
      {children}
    </AmplitudeProxy>
  </AnalyticsTrackingProvider>
);

export default AnaltyticsMockProvider;
