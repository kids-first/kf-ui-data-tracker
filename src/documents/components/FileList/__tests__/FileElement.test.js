import React from 'react';
import {MemoryRouter} from 'react-router-dom';
import {render, cleanup} from 'react-testing-library';
import FileElement from '../FileElement';
import {mocks} from '../../../../../__mocks__/kf-api-study-creator/mocks';
import studyByKfId from '../../../../../__mocks__/kf-api-study-creator/responses/studyByKfId';
import {MockedProvider} from 'react-apollo/test-utils';
import {AnalyticsProviderMock} from '../../../../analyticsTracking';

afterEach(cleanup);
it('renders correctly', () => {
  const file = studyByKfId.data.studyByKfId.files.edges[0].node;
  const tree = render(
    <AnalyticsProviderMock>
      <MockedProvider mocks={mocks}>
        <MemoryRouter>
          <table>
            <tbody>
              <FileElement
                fileListId={studyByKfId.data.studyByKfId.kfId}
                fileNode={file}
              />
            </tbody>
          </table>
        </MemoryRouter>
      </MockedProvider>
    </AnalyticsProviderMock>,
  );
  expect(tree.container).toMatchSnapshot();
});

it('renders loading state', () => {
  const file = studyByKfId.data.studyByKfId.files.edges[0].node;
  const tree = render(
    <AnalyticsProviderMock>
      <MockedProvider mocks={mocks}>
        <MemoryRouter>
          <table>
            <tbody>
              <FileElement
                loading={true}
                fileNode={file}
                fileListId={studyByKfId.data.studyByKfId.kfId}
              />
            </tbody>
          </table>
        </MemoryRouter>
      </MockedProvider>
    </AnalyticsProviderMock>,
  );
  expect(tree.container).toMatchSnapshot();
});

it('renders latest temporary state', () => {
  const file = studyByKfId.data.studyByKfId.files.edges[0].node;
  const DATE_TO_USE = new Date('2018-09-27T05:32:27+00:00');
  const _Date = Date;
  global.Date = jest.fn(() => DATE_TO_USE);
  global.Date.UTC = _Date.UTC;
  global.Date.parse = _Date.parse;
  global.Date.now = _Date.now;

  const tree = render(
    <AnalyticsProviderMock>
      <MockedProvider mocks={mocks}>
        <MemoryRouter>
          <table>
            <tbody>
              <FileElement
                loading={true}
                fileNode={file}
                fileListId={studyByKfId.data.studyByKfId.kfId}
              />
            </tbody>
          </table>
        </MemoryRouter>
      </MockedProvider>
    </AnalyticsProviderMock>,
  );
  expect(tree.container).toMatchSnapshot();
});
