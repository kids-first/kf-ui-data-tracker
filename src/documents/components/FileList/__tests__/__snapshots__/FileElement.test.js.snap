// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <table>
    <tbody>
      <tr
        class="cursor-pointer"
        data-testid="file-item"
        style="background-color: inherit;"
      >
        <td
          class="center aligned"
        >
          <div
            class="ui fitted checkbox"
            data-testid="file-select"
          >
            <input
              class="hidden"
              readonly=""
              tabindex="0"
              type="checkbox"
              value=""
            />
            <label />
          </div>
        </td>
        <td
          class="center aligned"
        >
          <i
            aria-hidden="true"
            class="ambulance icon"
          />
        </td>
        <td
          class=""
        >
          <i
            aria-hidden="true"
            class="info circle icon"
          />
          <a
            href="/study/undefined/documents/SF_5ZPEM167"
          >
            organization.jpeg
          </a>
        </td>
        <td
          class="center aligned four wide"
        >
          <div
            class="ui labels"
          >
            <span
              class="text-grey"
            />
          </div>
        </td>
        <td
          class="center aligned one wide"
        >
          <time
            datetime="2018-09-27T05:32:26.000Z"
            title="27 Sept 2018"
          >
            7 months ago
          </time>
        </td>
        <td
          class="center aligned"
        >
          <div
            class="ui small fluid buttons"
          >
            <button
              class="ui basic compact icon button"
            >
              <i
                aria-hidden="true"
                class="eye icon"
              />
            </button>
            <a
              class="ui basic compact icon button"
              data-testid="download-file"
              href="http://localhost:8080/download/study/SD_8WX8QQ06/file/SF_5ZPEM167/version/FV_KCDKCDYR"
              role="button"
            >
              <i
                aria-hidden="true"
                class="download icon"
              />
            </a>
            <button
              class="ui basic compact icon button"
              data-testid="copy-file-id"
            >
              <i
                aria-hidden="true"
                class="copy icon"
              />
            </button>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`renders latest temporary state 1`] = `
<div>
  <table>
    <tbody>
      <tr
        class="cursor-pointer"
        data-testid="file-item"
        style="background-color: inherit;"
      >
        <td
          class="center aligned"
        >
          <div
            class="ui fitted checkbox"
            data-testid="file-select"
          >
            <input
              class="hidden"
              readonly=""
              tabindex="0"
              type="checkbox"
              value=""
            />
            <label />
          </div>
        </td>
        <td
          class="center aligned"
        >
          <i
            aria-hidden="true"
            class="ambulance icon"
          />
        </td>
        <td
          class=""
        >
          <i
            aria-hidden="true"
            class="info circle icon"
          />
          <a
            href="/study/undefined/documents/SF_5ZPEM167"
          >
            organization.jpeg
          </a>
        </td>
        <td
          class="center aligned four wide"
        >
          <div
            class="ui labels"
          >
            <span
              class="text-grey"
            />
          </div>
        </td>
        <td
          class="center aligned one wide"
        >
          <time
            datetime="2018-09-27T05:32:27.000Z"
            title="27 Sept 2018"
          >
            7 months ago
          </time>
        </td>
        <td
          class="center aligned"
        >
          <div
            class="ui small fluid buttons"
          >
            <button
              class="ui basic compact icon button"
            >
              <i
                aria-hidden="true"
                class="eye icon"
              />
            </button>
            <a
              class="ui basic compact icon button"
              data-testid="download-file"
              href="http://localhost:8080/download/study/SD_8WX8QQ06/file/SF_5ZPEM167/version/FV_NVV741M1"
              role="button"
            >
              <i
                aria-hidden="true"
                class="download icon"
              />
            </a>
            <button
              class="ui basic compact icon button"
              data-testid="copy-file-id"
            >
              <i
                aria-hidden="true"
                class="copy icon"
              />
            </button>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`renders loading state 1`] = `
<div>
  <table>
    <tbody>
      <tr
        class="cursor-pointer"
        data-testid="file-item"
        style="background-color: inherit;"
      >
        <td
          class="center aligned"
        >
          <div
            class="ui fitted checkbox"
            data-testid="file-select"
          >
            <input
              class="hidden"
              readonly=""
              tabindex="0"
              type="checkbox"
              value=""
            />
            <label />
          </div>
        </td>
        <td
          class="center aligned"
        >
          <i
            aria-hidden="true"
            class="ambulance icon"
          />
        </td>
        <td
          class=""
        >
          <i
            aria-hidden="true"
            class="info circle icon"
          />
          <a
            href="/study/undefined/documents/SF_5ZPEM167"
          >
            organization.jpeg
          </a>
        </td>
        <td
          class="center aligned four wide"
        >
          <div
            class="ui labels"
          >
            <span
              class="text-grey"
            />
          </div>
        </td>
        <td
          class="center aligned one wide"
        >
          <time
            datetime="2018-09-27T05:32:26.000Z"
            title="27 Sept 2018"
          >
            7 months ago
          </time>
        </td>
        <td
          class="center aligned"
        >
          <div
            class="ui small fluid buttons"
          >
            <button
              class="ui basic compact icon button"
            >
              <i
                aria-hidden="true"
                class="eye icon"
              />
            </button>
            <a
              class="ui basic compact icon button"
              data-testid="download-file"
              href="http://localhost:8080/download/study/SD_8WX8QQ06/file/SF_5ZPEM167/version/FV_KCDKCDYR"
              role="button"
            >
              <i
                aria-hidden="true"
                class="download icon"
              />
            </a>
            <button
              class="ui basic compact icon button"
              data-testid="copy-file-id"
            >
              <i
                aria-hidden="true"
                class="copy icon"
              />
            </button>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
